event-app-api/merge_intervals/merge_intervals.test.js

```diff
@@ -0,0 +1,200 @@
+import { mergeIntervals, findGaps } from "./merge_intervals.js";
+
+describe("Merge Intervals", () => {
+  test("Events with overlapping times returns one start and end time", () => {
+    const testEvents = [
+      {
+        starttime: 1679990400000,
+        endtime: 1680076800000,
+      },
+      {
+        starttime: 1680076800000,
+        endtime: 1680235200000,
+      },
+      {
+        starttime: 1680166400000,
+        endtime: 1683772800000,
+      },
+    ];
+    const expectedResult = [
+      {
+        starttime: 1679990400000,
+        endtime: 1683772800000,
+      },
+    ];
+
+    expect(mergeIntervals(testEvents)).toEqual(expectedResult);
+  });
+
+  test("Events with non-overlapping events returns both of the events", () => {
+    const testEvents2 = [
+      {
+        starttime: 1679930880000,
+        endtime: 1679952000000,
+      },
+      {
+        starttime: 1679952100000,
+        endtime: 1679973600000,
+      },
+    ];
+
+    const expectedResult = [
+      {
+        starttime: 1679930880000,
+        endtime: 1679952000000,
+      },
+      {
+        starttime: 1679952100000,
+        endtime: 1679973600000,
+      },
+    ];
+
+    expect(mergeIntervals(testEvents2)).toEqual(expectedResult);
+  });
+
+  test("Events with the same startime and endttime returns both of the events", () => {
+    const testEvents3 = [
+      {
+        starttime: 1629816200000,
+        endtime: 1629816200000,
+      },
+      {
+        starttime: 1630002600000,
+        endtime: 1630002600000,
+      },
+    ];
+
+    const expectedResult = [
+      {
+        starttime: 1629816200000,
+        endtime: 1629816200000,
+      },
+      {
+        starttime: 1630002600000,
+        endtime: 1630002600000,
+      },
+    ];
+
+    expect(mergeIntervals(testEvents3)).toEqual(expectedResult);
+  });
+  test("Events with various date and time combination returns both of the events", () => {
+    const testEvents4 = [
+      {
+        starttime: 1679930880000,
+        endtime: 1680000000000,
+      },
+      {
+        starttime: 1685215200000,
+        endtime: 1685289600000,
+      },
+    ];
+
+    const expectedResult = [
+      {
+        starttime: 1679930880000,
+        endtime: 1680000000000,
+      },
+      {
+        starttime: 1685215200000,
+        endtime: 1685289600000,
+      },
+    ];
+
+    expect(mergeIntervals(testEvents4)).toEqual(expectedResult);
+  });
+  test("No events returns an empty array", () => {
+    const testEvents5 = [];
+
+    const expectedResult = [];
+
+    expect(mergeIntervals(testEvents5)).toEqual(expectedResult);
+  });
+});
+
+describe("Find gaps", () => {
```

Comment:
Add a test for when you have an event that ends at 9pm and another event that starts at 9pm so no gaps

Add a test for when both events have the same start/end time (both start at 9pm, end at 10pm) so there are no gaps

Also it would be good to combine tests so that you identify multiple cases of gaps